tests/perf/renderingSettings.cpp

```diff
@@ -51,8 +52,12 @@ BOOST_AUTO_TEST_CASE(default_scene_benckmark)
     timer.stop();
     reference = timer.milliseconds();
 
+    auto& renderer = brayns.getEngine().getRenderer();
+
     // Shadows
-    params.getRenderingParameters().setShadowIntensity(1.f);
+    auto props = renderer.getPropertyMap();
+    props.updateProperty("shadows", 1.f);
+    renderer.updateProperties(props);
```

Comment:
I haven't been part of the technical discussions, but this first usage example does not convince me that the change is an improvement:
1) The code is longer
2) Addressing by string is error-prone
3) I'm not sure if type safety is guaranteed; I have the impression that props.updateProperty("shadows", "no_a_number"); would compile just fine, or am I wrong and that's the reason that I can see no test for that?
4) It seem to me that internal properties of the Ospray shaders are being exposed directly throughout the application all the way to the web interface, completely breaking the encapsulation. If a shader in Ospray changes, all user code that relies on that magic string has to be fixed / recompiled / redeployed.

Reply:
That's the entire point: custom shaders have custom attributes, so they can and should no longer be hardcoded in the core. If the shader changes, it should be directly reflected on the client side (btw, the shader will not change really, but there will be new shaders from any plugin/module).

With the schema in place, having the properties for a specific shader being exposed is doable for the clients. Used here w/o any schema checks which tells the valid names and types, this looks wrong, agreed.

I addressed point 3, where the updated value must have the same type that the property was registered with.

Reply:
OK I see... if the shaders are moved to plugins then then PropertyMap is a way to provide reflexion for arbitrary C++ classes... in that case the string keys make sense. Then yes, make sure the intended usage is well tested:
PropertyMap is filled by plugin and only existing properties can be updated by clients in a type safe-manner.